pkg/controller/engine.go

```diff
@@ -0,0 +1,204 @@
+/*
+Copyright 2020 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package controller provides utilties for working with controllers.
+package controller
```

Comment:
I have mixed feelings about this package name; I like how `controller.NewEngine` reads but it seems like it will frequently conflict with `controller-runtime/pkg/controller`.

Reply:
I prefer the symmetry despite the naming conflicts 👍

Reply:
I've so far worked around this by importing the `controller-runtime/pkg/controller` as `ctrlr` (which is close to our convention of importing `sigs.k8s.io/controller-runtime` as `ctrl`.